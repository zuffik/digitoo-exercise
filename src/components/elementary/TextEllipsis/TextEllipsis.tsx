import * as React from 'react';
import classNames from 'classnames';
import styles from './TextEllipsis.module.sass';

interface Props extends React.HtmlHTMLAttributes<HTMLSpanElement> {
  maxWidth?: number;
}

export const TextEllipsis: React.FC<Props> = (props: Props) => {
  const maxWidth = props.maxWidth || 'none';
  return (
    <span
      {...props}
      className={classNames(props.className, styles.text)}
      style={{...props.style, maxWidth: typeof maxWidth == 'number' ? maxWidth + 'px' : maxWidth}}
    />
  );
};
