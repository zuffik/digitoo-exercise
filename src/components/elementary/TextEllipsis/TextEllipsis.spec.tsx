import * as React from 'react';
import {render} from '@testing-library/react'
import {TextEllipsis} from "./TextEllipsis";

describe('<TextEllipsis/>', () => {
    it('should render', () => {
        const {asFragment} = render(<TextEllipsis/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
