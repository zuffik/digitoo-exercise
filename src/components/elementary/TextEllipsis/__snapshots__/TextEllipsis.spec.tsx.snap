// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextEllipsis/> should render 1`] = `
<DocumentFragment>
  <span
    class="text"
    style="max-width: none;"
  />
</DocumentFragment>
`;
